src/main/osd/osd_elements.c

```diff
@@ -1127,6 +1127,55 @@ static void osdElementEfficiency(osdElementParms_t *element)
         tfp_sprintf(element->buff, "----%c/%c", SYM_MAH, unitSymbol);
     }
 }
+
+int16_t osdGetBoundaryFromDisplayX(int16_t error_x)
+{
+    if (error_x < -(osdConfig()->camera_fov_h / 2)){
+        error_x = -(osdConfig()->camera_fov_h / 2);
+    } else if (error_x >= osdConfig()->camera_fov_h / 2){
+        error_x = osdConfig()->camera_fov_h / 2;
+    }
```

Comment:
You can simplify the above 5 lines to only 2:

```suggestion
    const int fov2 = SIGN(error) * fov / 2;
    error = constrain(error, -fov2, fov2);
```